test/stage-options.test.ts

```diff
@@ -202,56 +217,281 @@ test('can set pre/post github action job step', () => {
-            with: {
-              'app-id': 4321,
-              'secrets': 'secrets',
+          ],
+        }),
+      ],
+
+      post: [
+        new GitHubActionStep('PostDeployAction', {
+          jobSteps: [
+            {
+              name: 'Checkout',
+              uses: 'actions/checkout@v3',
             },
-          },
-        ],
-      })],
+            {
+              name: 'post deploy action',
+              uses: 'my-post-deploy-action@1.0.0',
+              with: {
+                'app-id': 4321,
+                'secrets': 'secrets',
+              },
+            },
+          ],
+        }),
+      ],
+    });
+
+    app.synth();
+
+    const workflowFileContents = readFileSync(pipeline.workflowPath, 'utf-8');
+    expect(workflowFileContents).toMatchSnapshot();
+    expect(workflowFileContents).toContain('my-pre-deploy-action@1.0.0');
+    expect(workflowFileContents).toContain('my-post-deploy-action@1.0.0');
+    expect(workflowFileContents).toContain('actions/checkout@v3');
+    expect(workflowFileContents).toContain(
+      'contains(fromJson(\'["push", "pull_request"]\'), github.event_name)',
+    );
+    expect(workflowFileContents).toContain(
+      "success() && contains(github.event.issue.labels.*.name, 'deploy')",
+    );
+  });
+});
+
+test('stages in github waves works', () => {
+  withTemporaryDirectory((dir) => {
+    const pipeline = new GitHubWorkflow(app, 'Pipeline', {
+      workflowPath: `${dir}/.github/workflows/deploy.yml`,
+      synth: new ShellStep('Build', {
+        installCommands: ['yarn'],
+        commands: ['yarn build'],
+      }),
+    });
+
+    const stageA = new Stage(app, 'MyStageA', {
+      env: { account: '111111111111', region: 'us-east-1' },
+    });
+
+    new Stack(stageA, 'MyStackA');
+
+    const wave = pipeline.addGitHubWave('MyWave');
+
+    const stageAOptions: AddGitHubStageOptions = {
+      jobSettings: {
+        if: "success() && contains(github.event.issue.labels.*.name, 'deployToA')",
+      },
+    };
+    wave.addStageWithGitHubOptions(stageA, stageAOptions);
+
+    const stageBOptions: GitHubStageProps = {
+      env: { account: '12345678901', region: 'us-east-1' },
+      jobSettings: {
+        if: "success() && contains(github.event.issue.labels.*.name, 'deployToB')",
+      },
+    };
+    const stageB = new GitHubStage(app, 'MyStageB', stageBOptions);
+
+    new Stack(stageB, 'MyStackB');
+
+    wave.addStage(stageB);
+
+    app.synth();
+
+    const workflowFileContents = readFileSync(pipeline.workflowPath, 'utf-8');
+    expect(workflowFileContents).toMatchSnapshot();
+
+    const yaml = YAML.parse(workflowFileContents);
+    expect(yaml).toMatchObject({
+      jobs: {
+        'MyWave-MyStageA-MyStackA-Deploy': {
+          if: stageAOptions.jobSettings?.if,
+        },
+        'MyWave-MyStageB-MyStackB-Deploy': {
+          if: stageBOptions.jobSettings?.if,
+        },
+      },
+    });
+  });
+});
+
+test('github stages in waves works', () => {
```

Comment:
can we add a test for throwing an error if you try to add a stage after it is built?

Reply:
:+1: